includes/Core/Authentication/Authentication.php

```diff
@@ -980,12 +980,41 @@ private function get_reconnect_after_url_mismatch_notice() {
 			'reconnect_after_url_mismatch',
 			array(
 				'content'         => function() {
-					return sprintf(
+					$previous_connected_url = $this->previous_connected_proxy_url->get();
+					$connected_url          = $this->connected_proxy_url->get();
+					$content                = sprintf(
 						'<p>%s <a href="%s">%s</a></p>',
 						esc_html__( 'Looks like the URL of your site has changed. In order to continue using Site Kit, you’ll need to reconnect, so that your plugin settings are updated with the new URL.', 'google-site-kit' ),
 						esc_url( $this->get_proxy_setup_url() ),
 						esc_html__( 'Reconnect', 'google-site-kit' )
 					);
+					if ( is_string( $previous_connected_url ) && is_string( $connected_url ) && $connected_url !== $previous_connected_url ) {
+						$content = sprintf(
+							'<p>%s <a href="%s">%s</a></p>',
+							sprintf(
+							/* translators: 1: Previous URL. 2: Current URL */
+								esc_html__( 'Looks like the URL of your site has changed. In order to continue using Site Kit, you’ll need to reconnect, so that your plugin settings are updated with the new URL. Old url was %1$s, new url is %2$s.', 'google-site-kit' ),
+								esc_url( $previous_connected_url ),
+								esc_url( $connected_url )
+							),
+							esc_url( $this->get_proxy_setup_url() ),
+							esc_html__( 'Reconnect', 'google-site-kit' )
+						);
+					} elseif ( is_string( $connected_url ) && false === $previous_connected_url ) {
+						$content = sprintf(
+							'<p>%s <a href="%s">%s</a></p>',
+							sprintf(
+							/* translators: 1: Previous URL. 2: Current URL */
+								esc_html__( 'Looks like the URL of your site has changed. In order to continue using Site Kit, you’ll need to reconnect, so that your plugin settings are updated with the new URL. Old url was %1$s, new url is %2$s.', 'google-site-kit' ),
+								esc_url( $connected_url ),
+								esc_url( get_home_url() )
+							),
+							esc_url( $this->get_proxy_setup_url() ),
+							esc_html__( 'Reconnect', 'google-site-kit' )
+						);
+					}
```

Comment:
This can be a lot simplified by going with my suggestion from https://github.com/google/site-kit-wp/issues/4247#issuecomment-995226116. We don't need to entirely separate conditions, we can keep the `p` tag the same no matter what. Then, conditionally only append the `ul` list with the old and new URL.

Reply:
Yes, I can make that update, but the question in the ticket is more around the logic for which URLs to show.  In JS, we only show a message if there is a value set for both connected url and previous connected url https://github.com/google/site-kit-wp/blob/681fc336694ab2cc15221c6952e24792c75b17e0/assets/js/components/setup/SetupUsingProxy.js#L286-L293 

In PHP the approach covers an additional scenario:

> The issue we could now face is that I update my site kit plugin with the new version which introduces Previous Connected Proxy URL class, but has no value set, I then change my URL via the above filter, and get to a scenario where the url has changed, that previous_connected_url is not set (false), hence this alternate codepath

Reply:
I've pushed a few commits to simplify things here as it turns out the previous connected proxy URL wasn't actually needed as this is what the connected proxy URL would have. We always have access to the current URL (connected or not).